src/types.rs

```diff
@@ -1,11 +1,42 @@
-#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+use std::fmt::{Debug, Error, Formatter};
+
+#[derive(Clone, Eq, Hash, PartialEq)]
 pub struct Bytes(pub Vec<u8>);
 
-impl From<Vec<u8>> for Bytes {
-    fn from(t: Vec<u8>) -> Self {
-        Bytes(t)
+impl<T> From<T> for Bytes
+where
+    T: AsRef<[u8]>,
+{
+    fn from(t: T) -> Self {
+        Bytes(Vec::from(t.as_ref()))
     }
 }
 
 pub type Key = Bytes;
 pub type Value = Bytes;
+
+impl Debug for Bytes {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        write!(f, "{}", String::from_utf8_lossy(&self.0))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::Bytes;
+
+    #[test]
+    fn from() {
```

Comment:
If possible could these names be a little longer and more verbose? Especially the `debug` one below, it doesn't really tell me a lot about what it's doing as a function. I'm not sure what the Rust conventions are, but I might expect something like `it_converts_from_...`.